opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import TrainingArguments, get_linear_schedule_with_warmup
+
+
+class SentenceTransformerModel:
+    def __init__(self, model_url: str = None) -> None:
+        """
+        Description:
+        Initiate a sentence transformer model object.
+
+        Parameters:
+        model_url: str = None
+             the url to download sentence transformer model, if None, default as 'sentence-transformers/msmarco-distilbert-base-tas-b'
+        Return:
+             None
+        """
+        if model_url is None:
+            self.model_url = "sentence-transformers/msmarco-distilbert-base-tas-b"
+        else:
+            self.model_url = model_url
+
+    def train(
+        self,
+        read_path: str = None,
+        output_model_name: str = None,
+        output_model_path: str = None,
+        zip_file_name: str = None,
+        use_accelerate: bool = False,
+        compute_environment: str = None,
+        num_machines: int = 1,
+        num_processes: int = None,
+        learning_rate: float = 2e-5,
+    ) -> None:
+        """
+        Description:
+        Read the synthetic queries and use it to fine tune/train (and save) a sentence transformer model.
+
+        Parameters:
+        read_path: str
+            required, path to the zipped file that contains generated queries, if None, raise exception.
+            the zipped file should contain pickled file in list of dictionary format with key named as 'query',
+            'probability' and 'passages'. For example: [{'query':q1,'probability': p1,'passages': pa1}, ...].
+            'probability' is not required for training purpose.
+        output_model_path: str=None
+            the path to store trained custom model. If None, default as current folder path
+        output_model_name: str=None
+            the name of the trained custom model. If None, default as 'trained_model.pt'
+        zip_file_name: str =None
+            Optional, file name for zip file. if None, default as custom_tasb_model.zip
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator.
+        compute_environment: str
+            optional, compute environment type to run model, if None, default using 'LOCAL_MACHINE'
+        num_machines: int
+            optional, number of machine to run model , if None, default using 1
+        num_processes: int
+            optional, number of processors to run model , if None, default using 1
+        learning_rate: float
+            optional, learning rate to train model, default is 2e-5
+        Return:
+            None
+        """
+
+        query_df = self.read_queries(read_path)
+
+        train_examples = self.load_sentence_transformer_example(
+            query_df, use_accelerate
+        )
+
+        # if use accelerator to train model, run auto setuo accelerate confi and launch train_model function
```

Comment:
@dhrubo-os will this part be included when build python doc? I guess not, how about move to the train method comment, line 62?

Reply:
That's a good point. Yeah this part will not be included when build python doc. Better to add in the docString portion.

Reply:
I talked with @mingshl earlier. She is going to work in another PR targeting unit test and documentation.